pylint/checkers/base.py

```diff
@@ -1467,6 +1472,19 @@ def visit_dict(self, node: nodes.Dict) -> None:
                 self.add_message("duplicate-key", node=node, args=key)
             keys.add(key)
 
+    @utils.check_messages("duplicate-value")
+    def visit_set(self, node: nodes.Set) -> None:
```

Comment:
Can we add a `visit_call` to pick up the `set([1, 2, 2, 3])` initialization syntax?

Reply:
We might want to move that to a follow-up issue. I can see that becoming quite difficult and could inflate the size (and review burden) of this PR.

Reply:
Because of inferring iterables?

I was thinking just checking for `set` called on a list literal and then running the logic of `visit_set` on that list would capture a really common pattern and feasible for this PR.

Reply:
That sounds like a good check to add, but it's distinct from this one.

If someone writes `set([1, 2, 2, 3])`, they could get a hint to use set notation instead: `{1, 2, 2, 3}`. The next run would then trigger this check and suggest getting rid of the duplicate.

Maybe that could be collapsed to get the optimal result the first time, but I agree that it's outside the scope of this PR.

Reply:
Ah! Okay, my thinking was that `set([items])` and `{items}` are both supported ways of initializing a set. The [Python docs](https://docs.python.org/3/tutorial/datastructures.html#sets) aren't a style guide but are unopiniated on initializing sets: "Curly braces or the [set()](https://docs.python.org/3/library/stdtypes.html#set) function can be used to create sets."

The default check that prefers `{}` over `dict()` is also supported by the docs: "A pair of braces creates an empty dictionary: {}."

So my preference is to support both, but if we want to point people to `{items}` and not extend this PR to `set` I'll resolve this comment.

Reply:
Yeah, I think we should definitely support it, but in terms of reviewing it's just a lot easier to do that as a follow up.